homeassistant/components/fan/mqtt.py

```diff
@@ -259,19 +278,27 @@ def set_speed(self, speed: str) -> None:
                 mqtt_payload = self._payload[SPEED_HIGH]
             else:
                 mqtt_payload = speed
+
             self._speed = speed
```

Comment:
We should only do this and `self.hass.async_add_job(self.async_update_ha_state())` if `self._optimistic_speed`